src/assets/scripts/client/canvas/CanvasController.js

```diff
@@ -713,84 +717,143 @@ export default class CanvasController {
+            const fixPosition = NavigationLibrary.getFixRelativePosition(fix);
 
-                    textAtPoint[exitName].push(`${sid.identifier}.${exitName}`);
-                }
+            this._drawLabelsAtPoint(cc, fixPosition, textItemsToPrint);
+        }
+
+        cc.restore();
+    }
+
+    /**
+     * @for CanvasController
+     * @method _drawStars
+     * @param cc {HTMLCanvasContext}
+     * @private
+     */
+    _drawStars(cc) {
+        if (!this._shouldDrawStarMap) {
+            return;
+        }
+
+        const starLines = NavigationLibrary.getProcedureLines(PROCEDURE_TYPE.STAR);
+        const textAtFix = [];
+
+        cc.save();
+        cc.translate(CanvasStageModel.halfWidth, CanvasStageModel.halfHeight);
+        cc.strokeStyle = this.theme.SCOPE.STAR;
+        cc.fillStyle = this.theme.SCOPE.STAR;
+        cc.setLineDash([1, 10]);
+        cc.font = 'italic 14px monoOne, monospace';
+
+        for (let i = 0; i < starLines.length; i++) {
+            const star = starLines[i];
+
+            for (let j = 0; j < star.lines.length; j++) {
+                this._drawLine(cc, star.lines[j]);
             }
 
-            if (shouldDrawProcedureName) {
-                if (!(mostRecentFixName in textAtPoint)) {
-                    textAtPoint[mostRecentFixName] = [];
-                }
+            const lastFixName = star.lastFixName;
 
-                textAtPoint[mostRecentFixName].push(sid.identifier);
+            if (!(lastFixName in textAtFix)) {
+                textAtFix[lastFixName] = [];
             }
+
+            textAtFix[lastFixName].push(star.identifier);
         }
 
         // draw SID labels
-        for (const j in textAtPoint) {
-            const textItemsToPrint = textAtPoint[j];
-            const fixPosition = NavigationLibrary.getFixRelativePosition(j);
-            const fixCanvasPosition = CanvasStageModel.translatePostionModelToRoundedCanvasPosition(fixPosition);
-
-            for (let k = 0; k < textItemsToPrint.length; k++) {
-                const textItem = textItemsToPrint[k];
-                const x_position = fixCanvasPosition.x + 10;
-                const y_position = fixCanvasPosition.y + (15 * k);
+        for (const fix in textAtFix) {
+            const textItemsToPrint = textAtFix[fix];
+            const fixPosition = NavigationLibrary.getFixRelativePosition(fix);
 
-                cc.fillText(textItem, x_position, y_position);
-            }
+            this._drawLabelsAtPoint(cc, fixPosition, textItemsToPrint);
         }
 
         cc.restore();
     }
 
+    /**
+     * @for CanvasController
+     * @method _drawLine
+     * @param cc {HTMLCanvasContext}
+     * @param points {array of array<number, number>} position coordinates (in km)
+     * @private
+     */
+    _drawLine(cc, points) {
+        if (points.length < 2) {
+            return;
+        }
+
+        const lineStartPosition = CanvasStageModel.translatePostionModelToRoundedCanvasPosition(points[0]);
```

Comment:
Have we _always_ been translating to the first point of the line? I know a translate is among the most expensive canvas operations, but don't remember if it's necessary in this case. We're already doing a `.moveTo()` the first point's coordinates before the subsequent `.lineTo()`s, so is it needed?

Reply:
Yes, line 746 in the original file (CanvasController)

Reply:
Looks like we did always do this, and there is no noticeable performance difference, so this should be just fine. 👍